var/spack/repos/builtin/packages/py-scikit-build-core/package.py

```diff
@@ -51,6 +77,8 @@ class PyScikitBuildCore(PythonPackage):
     depends_on("py-setuptools", type="test")
     depends_on("py-wheel", type="test")
 
+    depends_on("py-importlib_metadata", type=("build", "run"))
```

Comment:
Only depends on importlib-metadata for python through 3.7. Spack package has dashes, not underscores.
```suggestion
    depends_on("py-importlib-metadata", type=("build", "run"), when="^python@:3.7")
```

Reply:
Also, this dependency only after 0.3.1, so:
```suggestion
    depends_on("py-importlib-metadata", type=("build", "run"), when="@0.3.1: ^python@:3.7")
```

Reply:
Sorry, I meant `@0.3:` in the when above.

Reply:
I don't think you applied the second version though.

Reply:
No, I click the wrong one....